mlog/human/parser.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package human
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mattermost/mattermost-server/mlog"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type LogEntry struct {
+	Time    time.Time
+	Level   string
+	Message string
+	Caller  string
+	Fields  []mlog.Field
+}
+
+func ParseLogMessage(msg string) LogEntry {
+	result, err := parseLogMessage(msg)
+	if err != nil {
+		// If failed to parse, just output a LogEntry where all fields are blank, but Message is the original string
+		var result2 LogEntry
+		result2.Message = msg
+		return result2
+	}
+	return result
+}
+
+func parseLogMessage(msg string) (result LogEntry, err error) {
+
+	dec := json.NewDecoder(strings.NewReader(msg))
+
+	// look for an initial "{"
+	if token, err := dec.Token(); err != nil {
```

Comment:
Instead of this parsing loop, would it be cleaner to decode the entry into a `map[string]interface{}` and go from there?

Reply:
The reason this parsing loop is here is because I wanted to preserve the order the fields were initially output in. Simply parsing the object into a map resulted in non-deterministic output. Although when I hooked the results up to logrus, I noticed they sort the fields alphabetically, so if that's acceptable we can simplify this a bit.

Reply:
I don't think the order matters. 0/5 though.

Reply:
For now I've just added a comment to the code noting the reason for the complexity and that could simplify if we want to.

Reply:
Looks good.